Geometry/TrackerNumberingBuilder/plugins/CmsTrackerLevelBuilder.cc

```diff
@@ -137,32 +137,62 @@ bool CmsTrackerLevelBuilderHelper::isLessRModule(const GeometricDet* a, const Ge
 
 bool CmsTrackerLevelBuilderHelper::isLessR(const GeometricDet* a, const GeometricDet* b) { return a->rho() < b->rho(); }
 
-template <class FilteredView>
-void CmsTrackerLevelBuilder<FilteredView>::build(FilteredView& fv,
-                                                 GeometricDet* tracker,
-                                                 const std::string& attribute) {
-  edm::LogVerbatim("TrackerGeometryBuilder") << "CmsTrackerLevelBuilder::build "
-                                             << " Building: " << fv.geoHistory();
-
-  edm::LogVerbatim("TrackerGeometryBuilder") << ExtractStringFromDDD<FilteredView>::getString(attribute, &fv) << " "
-                                             << tracker->type() << " " << tracker->name() << std::endl;
+template <>
+void CmsTrackerLevelBuilder<cms::DDFilteredView>::buildLoop(cms::DDFilteredView& fv,
+                                                            GeometricDet* tracker,
+                                                            const std::string& attribute) {
+  edm::LogVerbatim("TrackerNumberingBuilder") << "LevelBuilder::buildLoop " << fv.geoHistory() << std::endl;
+  edm::LogVerbatim("TrackerNumberingBuilder") << "\tskipFirstChild? " << skipFirstChild << std::endl;
+  uint psize = fv.geoHistory().size();
+  while (skipFirstChild || fv.firstChild()) {
```

Comment:
The intention of `skipFirstChild` is to avoid the `fv.firstChild()` call. In the old DD `fv.nextSibling()` allowed us to break this cycle (without changing the `fv` node) and go up in the "execution tree" (not referring to the geometry tree)

Reply:
@vargasa - If one of these is `true`, the loop will continue. Shouldn't it be `&&`?

Reply:
Hi @ianna The purpose of it was to avoid the call to `firstChild`, the `break` statement is done in a comparison with the size of `geoHistory` (now the `level`) after this. If `skipFirstChild` is `true`, `firstChild` is not executed and the iterator position is not changed.